index.bs

```diff
@@ -535,167 +738,319 @@ To determine whether an |attribute| <dfn>contains a javascript: URL</dfn>:
+1. Set |sanitizer|'s [=Sanitizer/configuration=] to |configuration|.
+1. Return true.
+
+</div>
+
+## Canonicalize the Configuration ## {#configuration-canonicalize}
+
+The {{Sanitizer}} stores the [=Sanitizer/configuration=] in a canonical form, as this makes
+a number of processing steps easier.
+
+<div class=example>An {{SanitizerConfig/elements}} list `{elements: ["div"]}` gets stored as
+`{elements: [{name: "div", namespace: "http://www.w3.org/1999/xhtml"}]`).
+</div>
+
+<div algorithm>
+To <dfn for=Sanitizer>canonicalize the configuration</dfn> {{SanitizerConfig}} |configuration|
+with a [=boolean=] |allowCommentsAndDataAttributes|:
+
+Note: We assume that |configuration| is the result of [[WebIDL]] converting a JavaScript
+value to a {{SanitizerConfig}}.
+
+1. If neither |configuration|["{{SanitizerConfig/elements}}"] nor
+    |configuration|["{{SanitizerConfig/removeElements}}"] [=map/exist=], then [=map/set=]
+    |configuration|["{{SanitizerConfig/removeElements}}"] to &laquo; [] &raquo;.
+1. If neither |configuration|["{{SanitizerConfig/attributes}}"] nor
+    |configuration|["{{SanitizerConfig/removeAttributes}}"] [=map/exist=], then [=map/set=]
+    |configuration|["{{SanitizerConfig/removeAttributes}}"] to &laquo; [] &raquo;.
+1. If |configuration|["{{SanitizerConfig/elements}}"] [=map/exists=]:
+   1. [=list/iterate|For each=] |element| of |configuration|["{{SanitizerConfig/elements}}"] do:
+      1. Replace |element| with the result of [=canonicalize a sanitizer element with attributes=]
+          |element|.
+1. If |configuration|["{{SanitizerConfig/removeElements}}"] [=map/exists=]:
+   1. [=list/iterate|For each=] |element| of |configuration|["{{SanitizerConfig/removeElements}}"] do:
+      1. Replace |element| with the result of [=canonicalize a sanitizer element=]
+          |element|.
+1. If |configuration|["{{SanitizerConfig/replaceWithChildrenElements}}"] [=map/exists=]:
+   1. [=list/iterate|For each=] |element| of |configuration|["{{SanitizerConfig/replaceWithChildrenElements}}"] do:
+      1. Replace |element| with the result of [=canonicalize a sanitizer element=]
+          |element|.
+1. If |configuration|["{{SanitizerConfig/attributes}}"] [=map/exists=]:
+   1. [=list/iterate|For each=] |attribute| of |configuration|["{{SanitizerConfig/attributes}}"] do:
+      1. Replace |attribute| with the result of [=canonicalize a sanitizer attribute=]
+          |attribute|.
+1. If |configuration|["{{SanitizerConfig/removeAttributes}}"] [=map/exists=]:
+   1. [=list/iterate|For each=] |attribute| of |configuration|["{{SanitizerConfig/removeAttributes}}"] do:
+      1. Replace |attribute| with the result of [=canonicalize a sanitizer attribute=]
+          |attribute|.
+1. If |configuration|["{{SanitizerConfig/comments}}"] does not [=map/exist=], then [=map/set=]
+    |configuration|["{{SanitizerConfig/comments}}"] to |allowCommentsAndDataAttributes|.
+1. If |configuration|["{{SanitizerConfig/attributes}}"] [=map/exists=] and
```

Comment:
Shouldn't  `{ removeAttributes: [], dataAttributes: true }` be an error as well? I think we can't detect it, if we do it like this.

Reply:
[=valid=] now treats this as an error.